src/de.rs

```diff
@@ -344,6 +342,38 @@ impl<'de, 'a, R: Read> de::Deserializer<'de> for &'a mut Deserializer<R> {
             .expect("struct_fields is not empty");
         Ok(result)
     }
+
+    #[inline]
+    fn deserialize_tuple<V: Visitor<'de>>(self, _len: usize, visitor: V) -> Result<V::Value> {
```

Comment:
I think this logic needs to be handled in the deserialize_any codepath - we want this check to happen even if someone calls `deserialize_any` rather than `deserialize_tuple`.